sdk/src/main/java/com/mapbox/maps/renderer/gl/TextureRenderer.kt

```diff
@@ -0,0 +1,156 @@
+package com.mapbox.maps.renderer.gl
+
+import android.opengl.GLES20
+import com.mapbox.maps.renderer.gl.GlUtils.toFloatBuffer
+
+internal class TextureRenderer(
+  private val depth: Float = 0f
+) {
+
+  private var program: Int = 0
+  private var attributePosition: Int = 0
+  private var attributeTextureCoord: Int = 0
+  private var uniformTexture: Int = 0
+  private var vbo: IntArray = IntArray(2)
+
+  fun initialize() {
+    setupVbo(
+      vertexArray = floatArrayOf(
+        -1f, -1f, depth,
+        1f, -1f, depth,
+        -1f, 1f, depth,
+        1f, 1f, depth,
+      ),
+      textureArray = floatArrayOf(
+        0f, 0f,
+        1f, 0f,
+        0f, 1f,
+        1f, 1f,
+      )
+    )
+
+    val vertexShader = GlUtils.loadShader(
+      GLES20.GL_VERTEX_SHADER,
+      VERTEX_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    val fragmentShader = GlUtils.loadShader(
+      GLES20.GL_FRAGMENT_SHADER,
+      FRAGMENT_SHADER_CODE
+    ).also(GlUtils::checkCompileStatus)
+
+    program = GLES20.glCreateProgram().also {
+      GlUtils.checkError("glCreateProgram")
+      GLES20.glAttachShader(it, vertexShader)
+      GlUtils.checkError("glAttachShader")
+
+      GLES20.glAttachShader(it, fragmentShader)
+      GlUtils.checkError("glAttachShader")
+
+      GLES20.glLinkProgram(it)
+      GlUtils.checkError("glLinkProgram")
+    }
+    attributePosition = GLES20.glGetAttribLocation(program, "aPosition")
+    attributeTextureCoord = GLES20.glGetAttribLocation(program, "aTexCoord")
+
+    uniformTexture = GLES20.glGetUniformLocation(program, "uTexture")
+  }
+
+  fun render(textureID: Int) {
+    if (program == 0) {
+      initialize()
```

Comment:
pretty strange to - I'd rather throw an exception if for some render() is called without `initialize()` beforehand

Reply:
Such flow simulates lazy init, only when render is actually performed, responded offline